apps/evaluations/forms.py

```diff
@@ -622,3 +626,47 @@ def clean_name(self):
                 raise forms.ValidationError("A dataset with this name already exists in your team.")
 
         return name
+
+    def clean(self):
+        cleaned_data = super().clean()
+        mode = cleaned_data.get("mode")
+
+        if mode == "clone":
+            session_ids, filtered_session_ids = self._clean_clone()
+            cleaned_data["session_ids"] = session_ids
+            cleaned_data["filtered_session_ids"] = filtered_session_ids
+
+        return cleaned_data
+
+    def clone_messages_to_dataset(self, dataset: EvaluationDataset) -> int:
+        """
+        Clone messages from sessions into the existing dataset, avoiding duplicates.
+        Uses ChatMessage IDs for duplicate detection.
+        """
+        session_ids = self.cleaned_data.get("session_ids", set())
+        filtered_session_ids = self.cleaned_data.get("filtered_session_ids", set())
+
+        if not session_ids and not filtered_session_ids:
+            return 0
+
+        new_evaluation_messages = self._create_messages_from_sessions(session_ids, filtered_session_ids)
```

Comment:
What happens to these if they were already in the dataset? Do they get left dangling?